pkg/autoscaler/aggregation/aggregation.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2019 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aggregation
+
+import "time"
+
+// Accumulator is a function accumulating buckets and their time relative to "now".
+type Accumulator func(now time.Time, time time.Time, bucket float64Bucket)
+
+// Windowed filters all buckets older than the given window relative to "now".
+func Windowed(window time.Duration, acc Accumulator) Accumulator {
+	return func(now time.Time, time time.Time, bucket float64Bucket) {
```

Comment:
Do we care about equal time? Since after would benefit readability (no negative in the if).

Reply:
Changed to `YoungerThan`, I think it's clearer then.